ansible/roles/prometheus/tasks/main.yml

```diff
@@ -27,6 +27,7 @@
     owner: prometheus
     group: prometheus
     mode: "0400"
+  when: (ansible_play_hosts | sort) == (hostvars | sort)
```

Comment:
> Why not just make sure that our variables are always correctly configured for the target hosts ?

This doesn't work properly because the variables rely on facts and other data discovered by Ansible as part of the playbook. We could sort of work around the facts thing by using fact caching, but it's brittle and thus may not work properly all the time.

Some things such as the `wg_pubkey` we could, in theory, define manually, but then it removes a degree of autonomity of the playbook and we have to configure more and more things on our own.

We could also set `ignore_errors: {{ (ansible_play_hosts | sort) != (hostvars | sort) }}` as a way of marking "we know this will fail because we don't run with all hosts" but since we can't carry this information over to the playbook runner (as with e.g. `@pytest.mark.xfail(reason="Joe Banks is lactose intolerant")` in pytest I'm not in favour, since then you need to guess if the error might actually be something bad